engine/src/cmd/collision.cpp

```diff
@@ -156,96 +141,104 @@ void Collision::shouldApplyForceAndDealDamage(Unit* other_unit)
+    // turn damage for AI-AI collisions back on, and then we can remove these caps.
+    //value, in seconds of desired maximum recovery time
 
-    // Perfectly elastic result
-    Vector new_velocity = v1*(m1-m2)/(m1+m2) + v2*2*m2/(m1+m2);
 
-    // Apply ratio
-    new_velocity = new_velocity * elasticity;
 
-    // Apply force
-    unit->ApplyForce(new_velocity);
+    //scale max applicable force by unit's ability to restore orientation and velocity, then convert from frames to seconds
+    float  max_force = max_force_multiplier * (unit->limits.forward + unit->limits.retro + unit->limits.lateral + unit->limits.vertical) / (unit->sim_atom_multiplier * SIMULATION_ATOM);
+    float  max_torque = max_torque_multiplier * (unit->limits.yaw + unit->limits.pitch + unit->limits.roll) / (unit->sim_atom_multiplier * SIMULATION_ATOM);
+    float force_magnitude = force.Magnitude();
+    if (force_magnitude > max_force) {
+        force *= (max_force / force_magnitude);
+    }
+    if (torque_magnitude > max_torque) {
+        torque *= (max_torque / torque_magnitude);
+    }
+    
+    */
+
+    // Apply force at location, should just be one call to ApplyLocalTorque.... but there's a catch - our current moment of inertia units(as in SI units, not unit class) are a bit b0rked, so we need to do a workaround :-(
+    Vector torqueForce = force * (1 / (0.667 * unit->radial_size * unit->radial_size)); // use shell approximation
+    Vector remainingLinearforce = force - torqueForce;
+    unit->ApplyLocalTorque(torqueForce, location_local);
+    unit->ApplyForce(remainingLinearforce);
 }
 
 
-// Discussion - the original code ran this once for both units. This required a comprison of the units to find out which is smaller.
+/*
+* Correctness check to validate collision math - should be eventually moved to testing, rather than on critical path
+*/
+void Collision::validateCollision(const QVector& relative_velocity,
+    const Vector& normal1,
+    const QVector& location1_local,
+    const QVector& location2_local,
+    const QVector& v1_new,
+    const QVector& v2_new,
+    const QVector& w1_new,
+    const QVector& w2_new) {
+    //following two values should be identical. Due to FP math, should be nearly identical. Checks for both absolute and relative error, the former to shield slightly larger errors on very small values - can set tighter/looser bounds later
+    double RestorativeVelAlongNormal = -1 * (1 - configuration.physics.inelastic_scale) * relative_velocity.Dot(normal1);
+    double ResultantVelAlongNormal = ((v2_new + w2_new.Cross(location2_local)) - (v1_new + w1_new.Cross(location1_local))).Dot(normal1);
+    double normalizedError = abs(1.0 - ResultantVelAlongNormal / RestorativeVelAlongNormal);
+    double absoluteError = abs(ResultantVelAlongNormal - RestorativeVelAlongNormal);
+    if (absoluteError > 0.001 && normalizedError > 0.01) { // absolute error > 1 milimeter/second along collision normal AND normalized error > 1%
```

Comment:
would be nice if the fuzzy compare was abstracted to its own inline function for re-use

Reply:
Probably a good idea to do so later, make it an inline function with both a percent and absolute bound - where in the codebase do you think is the best place to have testing/logging/etc. adjacent functions specified?

Reply:
Not sure; right now, it could just site in a function above this one.